metal/interrupt.h

```diff
@@ -319,6 +323,35 @@ metal_interrupt_get_priority(struct metal_interrupt *controller, int id) {
     return controller->vtable->interrupt_get_priority(controller, id);
 }
 
+/*!
+ * @brief Set preemptive level and priority for a given interrupt ID
+ *
+ * Set the preemptive level and priority for a given interrupt ID.
+ *
+ * @param controller The handle for the interrupt controller
+ * @param id The interrupt ID to enable
+ * @param level The interrupt level and priority are encoded together
+ * @return 0 upon success
+ */
+__inline__ int
+metal_interrupt_set_preemptive_level(struct metal_interrupt *controller, int id,
+                                     unsigned int level) {
+    return controller->vtable->interrupt_set_preemptive_level(controller, id,
+                                                              level);
+}
+
+/*!
+ * @brief Get an interrupt preemptive level
+ * @param controller The handle for the interrupt controller
+ * @param id The interrupt ID to enable
+ * @return The interrupt level
+ */
+__inline__ unsigned int
+metal_interrupt_get_preemptive_level(struct metal_interrupt *controller,
+                                     int id) {
+    return controller->vtable->interrupt_get_preemptive_level(controller, id);
```

Comment:
I see that the CLIC vtable is getting this, but the CLINT isn't? There should be a NULL check to make sure that calls to this public API don't dereference a NULL pointer for CLINT instances.

Reply:
Oh, will add a stub function for Clint. Thanks